lib/src/main/scala/spinal/lib/ethernet/IPv4Header.scala

```diff
@@ -0,0 +1,36 @@
+package spinal.lib.ethernet
+
+import EthernetProtocolConstant._
+import spinal.core._
+
+import scala.collection.mutable
+
+object IPv4Header extends HeaderOperate {
+  def apply(array: Seq[Bits]): Seq[Bits] = {
+    val gen = IPv4Header()
+    this.generate(array, gen)
+  }
+  def unapply(header: Bits): (Seq[String], Seq[Bits]) = {
+    val extract = IPv4Header()
+    this.extract(header, extract)
+  }
+}
+
+case class IPv4Header() extends FrameHeader {
+  val frameFieldInit = mutable.LinkedHashMap(
+    "protocolVersion" -> IP_VERSION_WIDTH,
```

Comment:
An alternative of using String would be to use Object : 

```scala
trait Key
object Keys{
  val protocolVersion = new Key
  val internetHeaderLength = new Key
  ..
}
  val frameFieldInit = mutable.LinkedHashMap(
    Keys.protocolVersion -> IP_VERSION_WIDTH,
```

Reply:
Got it. It seems more reasonable.